core/src/main/java/hudson/slaves/ChannelPinger.java

```diff
@@ -86,28 +88,42 @@ public ChannelPinger() {
 
     @Override
     public void preOnline(Computer c, Channel channel, FilePath root, TaskListener listener)  {
-        install(channel);
+        SlaveComputer slaveComputer = null;
+        if (c instanceof SlaveComputer) {
+            slaveComputer = (SlaveComputer) c;
+        }
+        install(channel, slaveComputer);
     }
 
+    /**
+     * @deprecated Use {@link #install(Channel, SlaveComputer)} instead.
+     */
+    @Deprecated
     public void install(Channel channel) {
+        install(channel, null);
+    }
+
+    @VisibleForTesting
```

Comment:
Never used in jenkinsci group.

Reply:
I confirm it's being used in one of private-source plugins. I would not deprecate the old method since there is no replacement in public API

Reply:
OK. I am just wondering, is it used for `SlaveComputer` channel from master's side so it _can_ be migrated to new API?

Reply:
Nope, it's another Remoting channel type. In new API it would be possible to pass `null` for sure (would not make it worse), but our use-case would probably require a special callback support in API. In `jenkinsci` org there is a similar use-case in Maven Plugin, where we could need a custom Remoting connection termination logic as well, e.g. force termination of the build (CC @aheritier)

We could do it API for sure, but IMHO it's out of the scope of this PR. I would rather prefer t keep it in the current state (just without API deprecation) so we could easily backport it to 2.73.x.

Reply:
Alright, so let's un-deprecate the old call and keep the new one private so we can give the API design a proper though later.